web/viewer.js

```diff
@@ -211,8 +211,20 @@ function webViewerLoad() {
       // set various `AppOptions`, by dispatching an event once all viewer
       // files are loaded but *before* the viewer initialization has run.
       const event = document.createEvent("CustomEvent");
-      event.initCustomEvent("webviewerloaded", true, true, {});
-      document.dispatchEvent(event);
+      event.initCustomEvent("webviewerloaded", true, true, {
+        source: window,
+      });
+      try {
+        // Attempt to dispatch the event at the top-level `document`,
+        // in order to support cases where the viewer is embedded in
+        // a *dynamically* created <iframe> element.
+        top.document.dispatchEvent(event);
```

Comment:
I was wondering if dispatching this event to the embedding document would be more correct, i.e. `parent.document`. I'm pretty sure in the vast majority of cases the embedder will be `top`, so maybe it doesn't matter that much :-)